lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,293 @@
+from rest_framework.views import APIView
+
+from enrollment.views import ApiKeyPermissionMixIn, EnrollmentCrossDomainSessionAuth, EnrollmentUserThrottle
+from lms.djangoapps.instructor.enrollment import enroll_email, get_email_params, get_user_email_language, unenroll_email
+from lms.djangoapps.instructor.views.tools import get_student_from_identifier
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from openedx.core.djangoapps.user_api.preferences.api import update_user_preferences
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from student.forms import PasswordResetFormNoActive
+from student.models import UserProfile
+from student.views import create_account_with_params
+from xmodule.modulestore.django import modulestore
+
+log = logging.getLogger(__name__)
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    Create user account without password
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Create user account without password
+
+        Creates a user using mail, login and also name and surname.
+        Sets a random password and sends a user a message to change it.
+        """
+        data = request.data
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        email = request.data.get('email')
+        username = request.data.get('username')
+        # countries.by_name return country code or '' for else cases
+        country = countries.by_name(request.data.get('country', ''))
+        first_name = request.data.get('first_name', '')
+        last_name = request.data.get('last_name', '')
+        language = request.data.get('language')
+        if not username:
+            error_msg = "username is required parameter."
+        elif not email:
+            error_msg =  "email is required parameter."
+        elif not country:
+            error_msg = (
+                "Country is wrong or absent: {country}. For checking: Visit https://www.iso.org/obp . "
+                "Click the Country Codes radio option and click the search button."
+            ).format(country=request.data.get('country', ''))
+        elif language not in ['en', 'pt']:
+            error_msg = (
+                "language is wrong or absent: {language}. It must be:  pt (Portuguese) or en (English)"
+            ).format(language=language)
+        else:
+            error_msg = None
+
+        if error_msg:
+            return Response(
+                data={"error_message": error_msg},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            validate_slug(username)
```

Comment:
Why do we need this separate validation? username is validated during create_account_with_params process.

Reply:
My code at first check params and at next to create a user. If the method 'create_account_with_params' will change and will remove validation, my code will not crush. 
And my code doesn't do search user if username isn't valid.